cocos/renderer/CCRenderer.h

```diff
@@ -372,6 +377,21 @@ class CC_DLL Renderer
     };
     ScissorState _scissorState;
     
+    enum StateFlag {
+        DEPTH_TEST = 1 << 0,
+        DEPTH_WRITE = 1 << 1,
+        CULL_FACE = 1 << 2,
+    };
+
+    struct StateBlock{
+        uint32_t modifiedStates = 0;
+        bool depthTest = false;
+        bool depthWrite = false;
+        backend::CullMode  cullMode = backend::CullMode::BACK;
+    };
+
+    std::deque<StateBlock> _stateBlockStack;
```

Comment:
Need a stack here?

Reply:
GroupCommand can cause a recursive calling, that we need a stack to track states.